aws/data_source_aws_iam_role.go

```diff
@@ -21,6 +16,11 @@ func dataSourceAwsIAMRole() *schema.Resource {
 			"assume_role_policy_document": {
 				Type:     schema.TypeString,
 				Computed: true,
+				Removed:  "Use `assume_role_policy` instead",
```

Comment:
I think we should deprecate things in the first stage and then eventually phase them out after a few releases, in a separate PR.